### PR TITLE
Adds JSON and Secret Custom Settings

### PR DESCRIPTION
**Pull request by swainn**
Moved @romer8 's branch into Tethys repo. See #923 for review comments.

image

### CODE REVIEW

**Comment by swainn** [comment]
@sdc50 and @shawncrawley this is ready for review. We're waiting on conda-forge to merge conda-forge/staged-recipes#22096, so the tests are failing.

File: tethys_apps/migrations/0004_auto_20230223_1731.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Generated by Django 3.2.16 on 2023-02-23 17:31
```
Comment (sdc50) [inline tethys_apps/migrations/0004_auto_20230223_1731.py:1]: I would recommend deleting all of these migrations, rebasing this branch, setting up a new DB, and then re-create a single migration.

**Review by sdc50 — CHANGES_REQUESTED** [review]

File: tethys_apps/migrations/0004_auto_20230223_1731.py (hunk 1 above)
Comment (swainn) [reply tethys_apps/migrations/0004_auto_20230223_1731.py]: Done

**Review by shawncrawley — APPROVED** [review]
This all looks good from what I could gather. I largely skimmed the test files, but went slowly and in-depth into the actual implementation. Good stuff.

**Pull request merged**